scripts/version-utils.js

```diff
@@ -0,0 +1,158 @@
+      verArr[1] = Number(verArr[1]) + 1;
+      verArr[2] = 0;
+      return verArr.join('.');
+    case 3: // Major
+      if (isBeta) {
+        return currentVersion.split('-')[0];
+      }
+      verArr[0] = Number(verArr[0]) + 1;
+      verArr[1] = 0;
+      verArr[2] = 0;
+      return verArr.join('.');
+    case 4: // Beta (e.g., 3.0.0-beta.0)
+      if (isBeta) { // Already a beta version then just increase the build number at the end
+        verArr[3] = Number(verArr[3]) + 1;
+        return verArr.join('.');
+      }
+      verArr[0] = Number(verArr[0]) + 1;
+      verArr[1] = 0;
+      verArr[2] = 0;
+      return verArr.join('.') + '-beta.0';
+    default:
+      logger.error(`ERROR: Invalid input: ${versionIncrement}`);
+      return undefined;
+  }
+};
+
+// Add an entry for the new version in CHANGELOG.md
+const updateChangelog = (newVersion) => {
+  logger.log(`Updating CHANGELOG.md with a new release entry - ${newVersion}`);
+  const filePath = path.resolve(__dirname, '../CHANGELOG.md');
+  let changeLog = fs.readFileSync(filePath).toString();
+  const latestEntryIndex = changeLog.indexOf('## [');
+  const newEntry = [
+    [`## [${newVersion}] - ${new Date().toISOString().slice(0, 10)}`],
+    ['### Added'],
+    ['### Removed'],
+    ['### Changed'],
+    ['### Fixed'],
+    [''],
+  ].join('\n\n');
+  changeLog = changeLog.substring(0, latestEntryIndex) + newEntry + changeLog.substring(latestEntryIndex);
+  fs.writeFileSync(filePath, changeLog);
+};
+
+// Update the base branch to point to a new branch.
+// For example, base branch for release-2.x should be origin/release-2.x
+const updateBaseBranch = (branchName) => {
+  logger.log(`Updating the base branch in .base-branch to ${branchName}`);
+  const filePath = path.resolve(__dirname, '../.base-branch');
+  fs.writeFileSync(filePath, `origin/${branchName}`);
+}
+
+const versionBump = (option, branchName) => {
+  if (!option) {
+    logger.log('Choose one of the following options to bump the next version:');
+    logger.log('  1. Patch');
+    logger.log('  2. Minor');
+    logger.log('  3. Major');
+    logger.log('  4. Beta');
+    option = Number(prompt(''));
+  }
+  const newVersion = getNewVersion(currentVersion, option);
+  if (!newVersion) {
+    process.exit(1);
+  }
+
+  branchName = branchName ? branchName : `version-bump-${newVersion}`;
+
+  const prevReleaseBranch = !isPreRelease(currentVersion) && (option === 3 || option === 4)
+    ? `release-${currentVersion.split('.')[0]}.x`
+    : '';
+
+  logger.warn('Warning: you are bumping the version\n');
+  logger.warn(`  From: ${currentVersion}\n`);
+  logger.warn(`  To: ${newVersion}\n`);
+  if (prevReleaseBranch) {
+    logger.warn(`  This will also create ${prevReleaseBranch} branch.`);
+  }
+  shouldContinuePrompt();
+
+  if (prevReleaseBranch) {
+    const currentBranch = (spawnOrFail('git', [' branch --show-current'], { skipOutput: true })).trim();
+    spawnOrFail('git', [`checkout -b ${prevReleaseBranch}`]);
```

Comment:
Do we want to do a `git fetch --all` here to get all remote branches, in case if this `prevReleaseBranch` is not present locally?

Also, could you please once explain why we are updating the same branch? If this is to merge a fix after cutting the branch say some bug fix after testing then the commit message is a bit confusing.